src/main/entities/Study.java

```diff
@@ -328,21 +321,105 @@ public String getStudyType() {
+        if (this.studyUserManager.getParticipants().isEmpty()) {
             this.studyType = studyType;
+            if (studyType.equals("Randomized")) {
+                this.randomizationMethod = "Block";
+            } else {
+                this.randomizationMethod = "N/A";
+            }
+            return true;
+        }
+        return false;
+    }
+
+
+    /**
+     * Retrieve the randomization method of the study.
+     *
+     * @return the randomization method of the study.
+     */
+    public String getRandomizationMethod() {
+        return randomizationMethod;
+    }
+
+
+    /**
+     * Reset the randomization method of the study.
+     * Only allowed when there is no eligible participants in the study. Also only allowed when the study type is
+     * "Randomized".
+     * <p>
+     * Precondition: the randomization method of the study must be "Block", "Simple", or "Stratified".
+     *
+     * @param randomizationMethod the randomization method of the study
+     * @return whether the change is successful
+     */
+    public boolean setRandomizationMethod(String randomizationMethod) {
+        if (this.studyUserManager.getParticipants().isEmpty() && this.studyType.equals("Randomized")) {
+            this.randomizationMethod = randomizationMethod;
+            return true;
+        }
+        return false;
+    }
+
+
+    /**
+     * Retrieve the stratification method of the study.
+     *
+     * @return the stratification method of the study.
+     */
+    public String getStratetificationMethod() {
+        return stratetificationMethod;
+    }
+
+
+    /**
+     * Reset the stratification method of the study.
+     * Only allowed when there is no eligible participants in the study. Also only allowed when the study type is
+     * "Randomized" and the randomization method is "Stratified".
+     * <p>
+     * The stratification method of the study must be a valid variable name in the eligibility questionnaire.
+     *
+     * @param stratetificationMethod the stratification method of the study
+     * @return whether the change is successful
+     */
+    public boolean setStratetificationMethod(String stratetificationMethod) {
+        if (this.studyUserManager.getParticipants().isEmpty() && this.studyType.equals("Randomized") &&
+                this.randomizationMethod.equals("Stratified")) {
+            this.stratetificationMethod = stratetificationMethod;
             return true;
         }
         return false;
     }
 
 
+    /**
+     * Retrieve the description of the study.
+     *
+     * @return the description of the study.
+     */
+    public String getStudyDescription() {
+        return studyDescription;
+    }
+
+
+    /**
+     * Modify the description of the study.
+     *
+     * @param studyDescription the new description of the study.
+     */
+    public void setStudyDescription(String studyDescription) {
+        this.studyDescription = studyDescription;
```

Comment:
Can we guarantee that this description is the same as the description in the consent form?

Reply:
I think we do not need to guarantee that. It is up to the researcher to decide what they want to put here and the consent form. The descriptions in these two places usually serve different purposes, so we can make them separately stored.